storage/pkg/command/gateway.go

```diff
@@ -222,6 +228,118 @@ func rules(cfg *config.Config, logger log.Logger) map[string]map[string]interfac
+	}
+	type mimetypesConfig map[string]mimeTypeConfig
+
+	var mimetypes mimetypesConfig
+
+	// load default app mimetypes from a json file
+	if cfg.Reva.AppRegistry.MimetypesJSON != "" {
+		data, err := ioutil.ReadFile(cfg.Reva.AppRegistry.MimetypesJSON)
+		if err != nil {
+			logger.Error().Err(err).Msg("Failed to read app registry mimetypes from JSON file: " + cfg.Reva.AppRegistry.MimetypesJSON)
+			return nil
+		}
+		if err = json.Unmarshal(data, &mimetypes); err != nil {
+			logger.Error().Err(err).Msg("Failed to unmarshal storage registry rules")
+			return nil
+		}
+		if err := mapstructure.Decode(mimetypes, &m); err != nil {
+			return nil
+		}
+		return m
+	}
+
+	logger.Info().Msg("No app registry mimetypes JSON file provided, loading default configuration")
+
+	mimetypes = map[string]mimeTypeConfig{
+		"application/pdf": {
+			Extension:   "pdf",
+			Name:        "PDF",
+			Description: "PDF document",
+			Icon:        "",
+			DefaultApp:  "",
+		},
+		"application/vnd.oasis.opendocument.text": {
+			Extension:   "odt",
+			Name:        "OpenDocument",
+			Description: "OpenDocument text document",
+			Icon:        "",
+			DefaultApp:  "",
+		},
+		"application/vnd.oasis.opendocument.spreadsheet": {
+			Extension:   "ods",
+			Name:        "OpenSpreadsheet",
+			Description: "OpenDocument spreadsheet document",
+			Icon:        "",
+			DefaultApp:  "",
+		},
+		"application/vnd.oasis.opendocument.presentation": {
+			Extension:   "odp",
+			Name:        "OpenPresentation",
+			Description: "OpenDocument presentation document",
+			Icon:        "",
+			DefaultApp:  "",
+		},
+		"application/vnd.openxmlformats-officedocument.wordprocessingml.document": {
+			Extension:   "docx",
+			Name:        "Microsoft Word",
+			Description: "Microsoft Word document",
+			Icon:        "",
+			DefaultApp:  "",
+		},
+		"application/vnd.openxmlformats-officedocument.spreadsheetml.sheet": {
+			Extension:   "xlsx",
+			Name:        "Microsoft Excel",
+			Description: "Microsoft Excel document",
+			Icon:        "",
+			DefaultApp:  "",
+		},
+		"application/vnd.openxmlformats-officedocument.presentationml.presentation": {
+			Extension:   "pptx",
+			Name:        "Microsoft PowerPoint",
+			Description: "Microsoft PowerPoint document",
+			Icon:        "",
+			DefaultApp:  "",
+		},
+		"application/vnd.jupyter": {
+			Extension:   "ipynb",
+			Name:        "Jupyter Notebook",
+			Description: "Jupyter Notebook",
+			Icon:        "",
+			DefaultApp:  "",
+		},
+		"text/markdown": {
+			Extension:   "md",
+			Name:        "Markdown file",
+			Description: "Markdown file",
+			Icon:        "",
+			DefaultApp:  "",
+		},
+		"application/compressed-markdown": {
+			Extension:   "zmd",
+			Name:        "Compressed markdown file",
+			Description: "Compressed markdown file",
+			Icon:        "",
+			DefaultApp:  "",
+		},
+	}
+
+	if err := mapstructure.Decode(mimetypes, &m); err != nil {
+		return nil
```

Comment:
that is a bit strange. How about an error return? 🙄

Reply:
I now added an error log, do you think that's sufficient?